Direct MAX7456 support - for AQ32 and future F4 targets

mlyle opened the pull request:
I WIP my hair back and forth.

Fixes #1332 

Large portions derived from https://github.com/UncleRus/MultiOSD
# General TODO
- [x] Doxygen for PIOS_MAX7456
- [x] Suck in and use more of the existing Max7456 driver 
- [x] PIOS_MAX7456 wait-for-vsync hack
- ~~Vsync interrupt support [defer, I think]~~
- [x] Meaningful content for widgets (@dustin is working on this some)
- [x] Font built into image
- [x] Paths cleanup
- [x] Font loading
- [x] Font selection (just have a couple built-ins) (@dustin)
- [x] Font tweaking
- [x] Splash / reset screen
- [x] Hook into Revolution (AirBot F4) target correctly
- [x] Hook into AQ32 target
- [x] Attributions/copyright notices tweaks
- [x] Disable Unusable Widgets [**blocker**]
- [x] Make sure default layout is somewhat usable.
- ~~Stats Screen [defer]~~
- [x] Confirm PlayUAV is OK.
# Widgets TODO
- [x] Alarms
- [x] Home Distance
- [x] Home Direction
- [x] Vario
- [x] Heading
- [x] Ground Speed
- [x] Air Speed
- [x] AHI

mlyle committed:
flight: remove printf2

Replace with library printf with floats.

mlyle committed:
pios_heap: have sbrk assert(0)

So that attempts to call it can be caught.

mlyle committed:
flight: pios_heap: provide malloc/free impl to c library

mlyle committed:
flight: changes to pios_heap for printfness

UncleRus committed:
Import initial character OSD code from MultiOSD

Some things we clearly won't use have been pruned out, though there's
the need for a lot more trimming and reorganization.

mlyle committed:
libcharosd: remove PROGMEM en masse

mlyle committed:
charosd: remove some other unused code

mlyle committed:
libcharosd: minimal copyright headers / context

mlyle committed:
pios_max7456: code complete on initial driver

other small factoring changes and tidyup of charosd stuff in the staging
area.

mlyle committed:
revolution: temporarily hack to put some chars on screen

mlyle committed:
pios_max7456: properly store spi id/slave idx

mlyle committed:
charosd: initial work on module (not complete)

mlyle committed:
revolution: character OSD outline almost complete

mlyle committed:
charosd: fixup to actually "work" minimally

mlyle committed:
pios_max7456: gross hack to poll vsync over SPI

mlyle committed:
playuavosd: fixup to work driver from multiosd

mlyle committed:
revolution: use library printf, with float

mlyle committed:
printf2: fixup remaining targets

mlyle committed:
pios_heap: space for sbrk needs to be WAY bigger

mlyle commented:
@tracernz --- if it's easy for you to confirm that I've not regressed playuavosd, it'd be appreciated (printf change, max7456 driver change).

dustin commented:
Do you want me to just push to this branch?  I tend to pull --rebase, so
pulling is clean for me.

On Sun, Sep 11, 2016, 21:16 Michael Lyle notifications@github.com wrote:

> @tracernz https://github.com/tracernz --- if it's easy for you to
> confirm that I've not regressed playuavosd, it'd be appreciated (printf
> change, max7456 driver change).
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/d-ronin/dRonin/pull/1345#issuecomment-246239759, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AAAG8_cH7Zmv20sotnn_4---1-7KQxmPks5qpNIDgaJpZM4J6ODC
> .

mlyle commented:
If you've got good edits, just push directly here, yes :D

dustin commented:
OK.  I don't have good ones.  But with a working printf, I can probably do
something pretty nice.

On Sun, Sep 11, 2016 at 9:38 PM Michael Lyle notifications@github.com
wrote:

> If you've got good edits, just push directly here, yes :D
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/d-ronin/dRonin/pull/1345#issuecomment-246243689, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AAAG8yW-dECj1dE0pBdBhjHbJpkKB1uQks5qpNcfgaJpZM4J6ODC
> .

tracernz commented:
This replaces original code that follows PiOS conventions (incl. Doxygen) written for this project with code copy-pasted from another project that doesn't conform. If we are going to do that I suggest keeping it in a board-specific dir until it is able to be brought up PiOS standard (or at least closer). :/

tracernz commented on flight/targets/revolution/fw/pios_config.h:
```diff
@@ -64,9 +64,15 @@
-#define PIOS_INCLUDE_RFM22B
-#define PIOS_INCLUDE_RFM22B_COM
-#define PIOS_INCLUDE_OPENLRS
+// XXX TODO DONTMERGE
```
Need to make sure this DOESN'T get in. :smiley:

mlyle replied:
it's on the todo list above :)

dustin committed:
Many standard panels filled in.

mlyle committed:
characterosd: add font data to characterosd module

mlyle committed:
charonscreendisplay: program font memory

mlyle committed:
pios_max7456: use a lot more of the existing driver

dustin committed:
Sexy splash screen

tracernz committed:
reintroduce @tracernz's excellent max7456 register map

Somewhat tweaked; no longer separate read / write register chunks, as
the read/write calls handle ORing addresses.

tracernz commented:
Got a board in the mail, this runs on it. :+1:

mlyle committed:
max7456: add vsync-guessing logic, clean up todos

mlyle committed:
max7456: address todos, fix header bugs

mlyle committed:
max7456: clarify ternary use

mlyle commented:
jenkins, test this please  (spurious 0xNone failure)

mlyle committed:
aq32: support for pios_max7456

mlyle commented:
jenkins, build artifacts please

mlyle commented:
@yamahabest --- I've pushed preliminary AQ32 OSD support-- completely untested.  Can you easily give it a try?

Should just be hooking up the OSD and setting the SPIPort to MAX7456 OSD.   Thx!

dustin committed:
charosd:  Throttle 0 -> 100%

dustin committed:
charosd: Added crosshair and panel at 0x0a

mlyle committed:
max7456: open-loop vsync per FRAME not field

dustin committed:
Added crosshair to font alternatives.

dustin committed:
Added font selection with up to three (3) fonts.

dustin committed:
Added names for all the special characters.

dustin committed:
Remove confusing and redundant symbols from panels

Lightning bolt in front of the flight time.
Light bulb in front of consumption.

Both of these are pretty obvious without misleading symbols.

dustin committed:
Alarm panel.

This is untested code at this point, but I primarily wanted to get the
uavo definition in so the compile hit would be out of the way.

mlyle committed:
revolution: allow user to select OSD

mlyle committed:
pios_rfm22b: validate device version

This way we're less eager to accept other SPI devices as a RFM22.

dustin committed:
Tell the user when the font changes.

mlyle committed:
revolution: fix missing break, whitespace

dustin committed:
Fix thin font loading.

mlyle committed:
systemmod: only touch rfm22 object if rfm22 running

Otherwise, the UAVO is probably not init'd and we'll assert blow.

mlyle committed:
charosd: initialize telemetry object

mlyle committed:
pios_max7456: add doxygen documentation

Also unexport some symbols for good luck.

dustin committed:
Center alarms if x > LINE_WIDTH (30)

mlyle commented on flight/Modules/CharacterOSD/panel.c:
```diff
@@ -461,16 +461,19 @@ STD_PANEL (AIRSPEED, 7, "\x88%d\x81", (int16_t) (/*XXX:telemetry::stable::airspe
 
 	uint8_t alarm_state;
 	AlarmString(&alarm, buffer, sizeof(buffer), false, &alarm_state);
+	if (x > LINE_WIDTH) {
```
would be nice to factor this behavior into pios_max7456_puts

dustin commented:
Yeah, I'll just do that.

On Wed, Sep 14, 2016 at 10:41 PM Michael Lyle notifications@github.com
wrote:

> ## _@mlyle_ commented on this pull request.
> 
> In flight/Modules/CharacterOSD/panel.c
> https://github.com/d-ronin/dRonin/pull/1345#pullrequestreview-96478:
> 
> >   uint8_t alarm_state;
> >   AlarmString(&alarm, buffer, sizeof(buffer), false, &alarm_state);
> > - if (x > LINE_WIDTH) {
> 
> would be nice to factor this behavior into pios_max7456_puts
> 
> —
> You are receiving this because you were mentioned.
> 
> Reply to this email directly, view it on GitHub
> https://github.com/d-ronin/dRonin/pull/1345#pullrequestreview-96478, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AAAG829bxYUKQJk-kkLEj727LA0dm0Pcks5qqNptgaJpZM4J6ODC
> .

dustin committed:
Refactor h centering into PIOS_MAX7456_puts

dustin committed:
Use MAX7456_FMT_H_CENTER for horizontal centering.

dustin committed:
Partial auto-disabling of unused panels.

This is sort of a sketch.  I think I'll move common checks into the
telemetry sourcing bits and then make these functions a bit lighter.
But this at least seems to work.

dustin committed:
Cheaper panel checking.

Calculate a bitmap of availble panels and use that both for gathering
information and deciding whether to display the panel.

dustin commented:
I wrote some code on the bus this morning that tracks panel availability with a bitmask as part of the loop and uses that for both determining whether to grab data as well as whether a panel is even available.  This should allow for dynamically turning stuff on just as the previous commit, but be significantly more efficient and somewhat easier to understand.

However, I've not pushed it yet because it could also completely not work.  I figured I'd test it before pushing it, but leave a note here in case someone gets the urge to make it better.

mlyle committed:
aq32: adjust SPI clock phasing

mlyle committed:
aq32: remove redundant spi id field

mlyle committed:
max7456: be sure to initialize the next_sync_expected field

yamahabest commented:
Tested on AQ32 and Sparkfun MAX7456. Started working after setting video format to PAL instead of NTSC. Probably setting is needed (Automatic detection/Fixed NTSC/Fixed PAL).

Also tried with the previous SPI clock phasing, seems to work also fine. Not sure what the change actually implies, and if new or previous is better.

mlyle commented on flight/Modules/CharacterOSD/charosd.h:
```diff
@@ -32,16 +32,29 @@ typedef struct {
+	int requirements;
 } panel_t;
 
+#define HAS_GPS		(1 << 0)
```
baro is an important one here

dustin commented:
I went with HAS_ALT for some reason.  I called the macro HAS_SENSORS, but
it's really more HAS_SENSE.  I figured ALT might come from multiple
locations and how we get it should be independent of whether it's displayed.

On Thu, Sep 15, 2016 at 8:38 PM Michael Lyle notifications@github.com
wrote:

> ## _@mlyle_ commented on this pull request.
> 
> In flight/Modules/CharacterOSD/charosd.h
> https://github.com/d-ronin/dRonin/pull/1345#pullrequestreview-290377:
> 
> >  } panel_t;
> 
> +#define HAS_GPS      (1 << 0)
> 
> baro is an important one here
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/d-ronin/dRonin/pull/1345#pullrequestreview-290377,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AAAG87siHTDKU9ZiMpV4A4kVtDKvtNJWks5qqg8ggaJpZM4J6ODC
> .

mlyle commented:
Woops, my bad. :)

dustin committed:
Detect stalled MAX7456 chip and reset all the things

dustin commented:
OK, some interesting stuff (and justification for my latest commit).

If I leave this thing plugged in in my quad and just have it sit idle, weird shit starts to happen:
- ADC voltage starts rising (e.g. 18V for my 4S battery)
- Starts failsafing (MAN:NO RX, also confirmed by the taranis losing contact)
- OSD will stop drawing.
- Spinning up the motors starts to get difficult (bad signal).

Once the motors start spinning, things go back to normal.

Theory:  FC gets hot and components stop componenting properly.
Test:  Wait for stuff to break down and just stick it front of a fan.  Things go back to normal.

The stall detection code also kicks in and brings back the OSD once things are cool enough.

dustin committed:
max7456 init:  name that bit

mlyle commented:
> ADC voltage starts rising (e.g. 18V for my 4S battery)

What this really indicates is that Vref (which is the microcontroller's VCC / 3V3) starts falling-- e.g. the regulator is going into thermal protection.  This probably means the 5V supply is sagging waaaaay down.  Not awesome.

dustin commented:
Right. All the stuff I saw seems to support that idea.  It'd be nice to get
a thermometer on the FC. To test this.  The osd itself has support for
displaying that. Heh

On Sat, Sep 17, 2016, 02:40 Michael Lyle notifications@github.com wrote:

> ADC voltage starts rising (e.g. 18V for my 4S battery)
> 
> What this really indicates is that Vref (which is the microcontroller's
> VCC / 3V3) starts falling-- e.g. the regulator is going into thermal
> protection. This probably means the 5V supply is sagging waaaaay down. Not
> awesome.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/d-ronin/dRonin/pull/1345#issuecomment-247759963, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AAAG889scmGY2q5sdiJFpoBHUlewTm9iks5qq7VrgaJpZM4J6ODC
> .

mlyle commented:
#define PIOS_ADC_USE_TEMP_SENSOR    may work ... it's only used on CC3D for some reason.

tracernz commented:
Yeah, we normally don't really care about the CPU temperature. It's just an ADC channel so should work if the code is there. :-)

mlyle commented:
We don't look at the right factory-calib-coeffs.. but maybe the scale rule there is generic enough to be mostly right (even across families).

dustin commented:
I don't think I would, but is the idea to just use a thermistor?

On Sat, Sep 17, 2016, 14:31 Michael Lyle notifications@github.com wrote:

> We don't look at the right factory-calib-coeffs.. but maybe the scale rule
> there is generic enough to be mostly right (even across families).
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/d-ronin/dRonin/pull/1345#issuecomment-247809326, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AAAG8_nGKgeP6cBa2OQG9zkwYdI3zW7vks5qrFwpgaJpZM4J6ODC
> .

mlyle commented:
The STM32 has an internal thermal reference (a "bad" bandgap voltage reference with a big tempco, I think).  We measure it with the ADC.  But the little bit of code that puts the temperature into systemstats isn't activated for anything but CC3D.

Not sure at all the calibration is right, but if it measures something around 30C at powerup it's probably good. 

Should just be a question of adding that define to pios_board.h -- and hopefully the rest of stuff works.

dustin commented:
Oh awesome.  I'll see if I can make this work later tonight.

On Sat, Sep 17, 2016 at 3:23 PM Michael Lyle notifications@github.com
wrote:

> The STM32 has an internal thermal reference (a "bad" bandgap voltage
> reference with a big tempco, I think). We measure it with the ADC. But the
> little bit of code that puts the temperature into systemstats isn't
> activated for anything but CC3D.
> 
> Not sure at all the calibration is right, but if it measures something
> around 30C at powerup it's probably good.
> 
> Should just be a question of adding that define to pios_board.h -- and
> hopefully the rest of stuff works.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/d-ronin/dRonin/pull/1345#issuecomment-247811767, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AAAG8x6o5XowujJc_Si-1XdvVvC9Iy-Dks5qrGhggaJpZM4J6ODC
> .

mlyle commented:
Linkin' the video from IRC so I don't lose it: http://gfycat.com/SinfulAdeptAmericanriverotter

yamahabest commented:
jenkins, build artifacts please

yamahabest commented:
Does it even work, when I comment that?

mlyle commented:
nope. ;)

jenkins, build artifacts please

yamahabest commented:
Ah, you have trained him well ;)

mlyle commented:
Looks like it's broken at the moment even for me.  Will look at it later if it stays broken.

mlyle commented:
jenkins, build artifacts please

yamahabest commented:
Ehm... not sure what is causing this, but with this latest build, after enabling the OSDPort, and applying power, my AQ32 starts bootlooping when external power is applied. The ESCs don't make their startup sound anymore, supporting that the PWM outputs are not initialized due to the bootloop.

Without OSD enabled, the green led is solid, the blue led blinks.
With OSD enabled, the green led blinks, and the blue led is short bright, long less bright.

OSD is outputting a grey image also when the OSD is enabled.
When the OSD is disabled, I see the camera feed.

dustin committed:
Functional AHI

dustin committed:
Fix init registration namings

mlyle commented:
jenkins, build artifacts please

yamahabest commented:
Yay, OSD is working again on AQ32.

mlyle committed:
aq32: use the correct printf impl

dustin committed:
Functional climb panel.

dustin committed:
Working ground speed.

mlyle commented:
jenkins, build artifacts please

dustin committed:
Grab all of AttitudeActual and PositionActual.

dustin committed:
Home direction.

dustin committed:
Distance home

dustin committed:
Enable compass panel if there's a compass.

dustin committed:
Apply yaw actual to home direction

dustin committed:
Added a heading panel

dustin committed:
Added an airspeed panel.

This isn't using STD_PANEL because I just don't bother getting the
airspeed except in the rare cases someone needs it.  There are other
panels in this form and this could likely be generalized, but this at
least prevents us from grabbing data that's almost never going to be
present.

dustin committed:
Remove commented out code and odd spaces.

dustin committed:
Cleaner, more consistenty working home direction

mlyle commented on flight/Modules/CharacterOSD/panel.c:
```diff
@@ -0,0 +1,471 @@
+	float home_dir = (atan2f(state->telemetry.position_actual.East,
+				 state->telemetry.position_actual.North) * RAD2DEG) + 180;
+
+	if (HAS_SENSOR(state->available, HAS_COMPASS)) {
```
should we do this, or should we suppress the thing entirely?  If there's no compass, it's suddenly pointing in compass-rose terms.

mlyle commented on flight/Modules/CharacterOSD/panel.c:
```diff
@@ -0,0 +1,471 @@
+
+const char _horz_line [PANEL_HORIZON_WIDTH + 1]   = "\xb8            \xb9";
+const char _horz_center [PANEL_HORIZON_WIDTH + 1] = "\xc8            \xc9";
+static double deg2rad(double deg) {
```
miscmath.h's DEG2RAD is a constant.  Also can these be floats instead of double?

mlyle commented on flight/Modules/CharacterOSD/panel.c:
```diff
@@ -0,0 +1,471 @@
+/* GroundSpeed */
+STD_PANEL(GROUNDSPEED, 7, "\x80%d\x81", (int)round(pythag(state->telemetry.velocity_actual.North,
+							  state->telemetry.velocity_actual.East)
+						   * 3.6));
```
is this meters/second to km/hr?

mlyle commented on flight/Modules/CharacterOSD/panel.c:
```diff
@@ -0,0 +1,471 @@
+
+	const char * const levels [] = { _l0, _l1, _l2, _l3, _l4, _l5 };
+
+	uint8_t level = round (state->telemetry.manual.rssi / 20.0);
```
20.0f

mlyle commented on flight/Modules/CharacterOSD/panel.c:
```diff
@@ -0,0 +1,471 @@
+
+	// code below from minoposd
+	int16_t pitch_line = tan(deg2rad(-state->telemetry.attitude_actual.Pitch)) * _PAN_HORZ_LINES;
+	float roll = tan(deg2rad(state->telemetry.attitude_actual.Roll));
```
tanf

mlyle commented on flight/Modules/CharacterOSD/panel.c:
```diff
@@ -0,0 +1,471 @@
+}
+
+/* Lat */
+STD_PANEL(LATITUDE, 11, "\x83%02.6f", (double)state->telemetry.gps_position.Latitude / 10000000.0);
```
ugh.  These are marginal/ should probably remain as doubles.  Or we could do fixed point explicitly (carefully)

mlyle commented on flight/Modules/CharacterOSD/panel.c:
```diff
@@ -0,0 +1,471 @@
+#include "systemalarms.h"
+#include "modulesettings.h"
+
+static inline double pythag(float a, float b) {
```
float, sqrtf

mlyle committed:
max7456/charosd: finetune attributions

mlyle committed:
charosd: fine tune floating point, style

tracernz reviewed (commented):
Thanks for closing those groups! :-D

mlyle commented:
Confirmed OK on playuav.

yamahabest commented:
Just wondering, is it part of this PR to be able to force video signal to PAL/NTSC?
The last time I tested, the code wasn't able to detect my video signal automatically (slow cam or so), and reverted to NTSC.
My camera is outputting PAL however, so that results in no overlay.
I could hack the code manually, but I could expect more people to face this issue.

I believe @dustin was facing the same issue (failed auto detection), but current code is working for him, as he has a cam that is outputting NTSC.

dustin commented:
I don't have an easy way to test PAL.  It's possible to have a setting to
force a standard in the uavo and have the code do that instead of
autodetect, but I don't know who great the testing will be.

On Sun, Sep 25, 2016 at 12:40 AM yamahabest notifications@github.com
wrote:

> Just wondering, is it part of this PR to be able to force video signal to
> PAL/NTSC?
> The last time I tested, the code wasn't able to detect my video signal
> automatically (slow cam or so), and reverted to NTSC.
> My camera is outputting PAL however, so that results in no overlay.
> I could hack the code manually, but I could expect more people to face
> this issue.
> 
> I believe @dustin https://github.com/dustin was facing the same issue
> (failed auto detection), but current code is working for him, as he has a
> cam that is outputting NTSC.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/d-ronin/dRonin/pull/1345#issuecomment-249407783, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AAAG82uDXvfeQf5AH5zyFUXydPVajb3Jks5qtiV0gaJpZM4J6ODC
> .

yamahabest commented:
Fine by me. I can test.
Perhaps 3 options? Automatic, Force to NTSC, Force to PAL?

mluessi commented:
Yeah, being able to set the video system is probably a good idea. Detection on frame rate alone may not be super reliable. IMO, it's weird that the max chip can't detect the system, it's pretty easy to implement by counting the number of lines.

mlyle commented:
MAX chip does autodetect.  Right now we peg things to NTSC if there's no video present at startup.

mlyle committed:
charonscreendisplaysettings: fix indent

mlyle committed:
charonscreendisplaysettings: add descriptions

mlyle committed:
charonscreendisplay: allow forcing vid standard

mlyle commented:
OK, @yamahabest, implemented what you wanted.  Can you test?

yamahabest commented:
Yes, working! You guys rock!

Some small comments, based on this screen shot: https://snag.gy/YCXSig.jpg
- The "2f" and "6f" have been solved on AQ32. Yay! (Y).
- The crosshair is not in the middle of the artificial horizon on PAL. I think I have seen some stuff posted on NTSC that did have the crosshair in the middle?
- There are some small artifacts on the top, and on the middle, but I am expecting everyone has those?
- I am not sure where those dashes are for, that are just below and in between the "A" and the "FAIL" text?

dustin commented:
The cross hair is wherever it's configured to be. There are more lines on
pal, so you'd need adjust. The default config is good for NTSC.

We talked about having a magic number for vertical centering which could
help here.

On Sun, Sep 25, 2016, 14:33 yamahabest notifications@github.com wrote:

> Yes, working! You guys rock!
> 
> Some small comments, based on this screen shot: https://snag.gy/YCXSig.jpg
> - The "2f" and "6f" have been solved on AQ32. Yay (Y).
> - The crosshair is not in the middle of the artificial horizon. I
>   think I have seen some stuff posted on NTSC that did have the crosshair in
>   the middle?
> - There are some small artifacts on the top, and on the middle, but I
>   am expecting everyone has those?
> - I am not sure where those dashes are for, that are just below and in
>   between the "A" and the "FAIL" text?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/d-ronin/dRonin/pull/1345#issuecomment-249448047, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AAAG86hHOP4HoY6hcrBQfgI3avENqjcFks5qtuizgaJpZM4J6ODC
> .

yamahabest commented:
Probably being dumb here, but where do I configure these things?

dustin commented:
The charosd uavo has panels defined in slots and each slot has an x and y
location.  31 is a magical x that means "center"

On Mon, Sep 26, 2016, 01:01 yamahabest notifications@github.com wrote:

> Probably being dumb here, but where do I configure these things?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/d-ronin/dRonin/pull/1345#issuecomment-249504121, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/AAAG84LZYGynajnBvv9NqzAVzlIlTkZ0ks5qt3vogaJpZM4J6ODC
> .

tracernz commented:
@yamahabest In the UAVO browser. There is no UI for this stuff.

mlyle commented:
Rebased to fix small conflicts:
- touched a file adjacent to pios_led.c in the revolution makefile
- Indentation changed in systemmod where we made the updating of the RFM22 status object conditional

mlyle commented:
Can this merge pretty please?

tracernz commented on shared/uavobjectdefinition/charonscreendisplaysettings.xml:
```diff
@@ -2,6 +2,16 @@
+		<field name="VideoStandard" units="enum" type="enum" elements="1" defaultvalue="AutodetectPreferPAL">
+			<description>The video standard to use for the OSD</description>
+			<options>
+				<option>AutodetectPreferPAL</option>
```
You can use spaces (+some other punctuation) in this to look nicer in UI, and it will be collapsed in the generated UAVO code. :stuck_out_tongue:

mlyle committed:
charosd: change options to be prettier

tracernz commented:
jenkins, build artifacts please 

Paranoid/don't want to break the tree at this stage in the cycle :-P
